advanced-javascript/tests/arrays.test.js

```diff
@@ -18,43 +18,86 @@ describe('Arrays', () => {
   describe('`each`', () => {
     it('should be a function', () => {
       const each = arrayFunctions.each;
-      expect(each).to.be.a('string');
+      expect(each).to.be.a('function');
     });
     // begin here
+    it('should invoke callback of the length of array', () => {
+      const each = arrayFunctions.each;
+      const callBack = sinon.spy();
+      each([1,2,3], callBack);
+      expect(callBack).to.have.callCount(3);
+    });
   });
 
   describe('`map`', () => {
     it('should be a function', () => {
       const map = arrayFunctions.map;
```

Comment:
Instead of declaring the `map` variable inside each `it()` statement, you can declare `map` once _outside_ the `it()` statement and reuse! (DRY — don't repeat your code!)

i.e.:
```
const { map } = arrayFunctions;

it('first', () => {
  expect(map)...
}
it('second', () => {
  expect(map)...
}
```